version: "2.1"

services:
  plex:
    # build:
    #   context: ./plex
    # network_mode: "host"
    ports:
      - 32400:32400/tcp
      - 3005:3005/tcp
      - 8324:8324/tcp
      - 32469:32469/tcp
      - 1900:1900/udp
      - 32410:32410/udp
      - 32412:32412/udp
      - 32413:32413/udp
      - 32414:32414/udp
    restart: unless-stopped
    environment:
      - PUID=${PLEX_PUID:-1000}
      - PGID=${PLEX_PGID:-1000}
      - TZ="Europe/Madrid"
      - ADVERTISE_IP="http://192.168.1.197:32400/"
    volumes:
      - plex-config-volume:/config
      - rtorrent-downloads-volume:/data:z
      - /etc/localtime:/etc/localtime:ro
    networks:
      - proxy-network
    labels:
      - 'traefik.enable=true'
      - 'traefik.port=32400'
      - 'traefik.frontend.rule=Host:plex.${DOMAIN}'
      - 'traefik.frontend.passHostHeader=true'
      # - 'traefik.frontend.auth.basic=${BASIC_AUTH}'
      - 'traefik.docker.network=host'
    healthcheck:
      test: nc -z 127.0.0.1 32400
      interval: 5s
      timeout: 10s
      retries: 7

volumes:
  plex-config-volume: {}
  rtorrent-downloads-volume:
    external: true

networks:
  proxy-network:
    external: true
